src/CPUActivityAgent.cpp

```diff
@@ -70,13 +71,34 @@ namespace geopm
 
     void CPUActivityAgent::init_platform_io(void)
     {
-        for (int domain_idx = 0; domain_idx < M_NUM_CORE; ++domain_idx) {
+        int scalability_signal_domain = m_platform_io.signal_domain_type("MSR::CPU_SCALABILITY_RATIO");
+        if (scalability_signal_domain < m_freq_ctl_domain_type) {
```

Comment:
I'm not sure whether this type of comparison is always expected to hold. `m_platform_topo.is_nested_domain` exposes that relationship at the interface level.

Reply:
Doing away with the if entirely could land us in a situation where scalability is per CPU but frequency control is per Core, ending in a different error from the FrequencyGovernor so the second approach is what was used via the if block, but it should be switched to use isnested and remove the if as suggested.